enhancements/network/allow-from-router-networkpolicy.md

```diff
@@ -0,0 +1,236 @@
+strategy, the NetworkPolicy-relevant behavior also varies depending on
+the network plugin in use. This makes it impossible to create an
+"allow traffic from routers" policy that will work in any cluster.
+Indeed, in some cases, it is difficult to create such a policy even
+just for a single cluster, even when you know exactly how the cluster
+is configured.
+
+## Motivation
+
+### Goals
+
+- Allow a user to create a policy which will allow traffic to their
+  pods from routers (and not from arbitrary pod-network pods), which
+  will work regardless of network plugin or router endpoint publishing
+  strategy.
+
+### Non-Goals
+
+- Allowing a user to create a policy which is 100% guaranteed to allow
+  traffic _only_ from routers (and, eg, not from other host-network
+  pods).
+
+## Proposal
+
+### Background
+
+When the router is run with the `NodePortService` or
+`LoadBalancerService` endpoint publishing strategy, the router pods
+run on the pod network, and so router traffic can be allowed by
+creating NetworkPolicies that allow traffic based on an appropriate
+`namespaceSelector`. We currently label the `openshift-ingress`
+namespace with the label `network.openshift.io/policy-group: ingress`
+for exactly this purpose.
+
+However, when the router is run with the `HostNetwork` endpoint
+publishing strategy, the router pods will be on the host network, and
+so their traffic will not be recognized as coming from any specific
+namespace.
+
+In OpenShift SDN, for somewhat accidental historical reasons, traffic
+coming from any non-pod source is treated as though it came from the
+namespace "`default`". Thus, it's possible to allow traffic from
+routers by adding a label to `default` and then allowing traffic based
+on that label. (This will also allow traffic from any other
+host-network pod, or node process.) However, this approach does not
+work with any other network plugin.
+
+In theory, with other plugins, if you knew the pod network CIDR, you
+could create a policy saying "allow from anywhere (that can route to
+the pod network) except the pod network", which is essentially
+equivalent to "allow from `default`" in OpenShift SDN. eg:
+
+    ipBlock:
+      cidr: 0.0.0.0/0
+      except:
+        - 10.128.0.0/14
+
+This is not ideal, because it requires you to know the pod network
+CIDR, but even given that, it would not actually work with many
+network plugins anyway. Eg, in ovn-kubernetes, when a node or a
+host-network pod sends traffic to a pod-network pod, the source IP is
+not the node's "primary" node IP but rather the IP associated with the
+`ovn-k8s-mp0` interface, which is inside the pod network CIDR (eg, it
+would be `10.128.6.2` on the node that has the `10.128.6.0/23`
+subnet).
+
+It would be possible to create a policy allowing specifically the
+relevant source node IP of each node, but such a policy would need to
+be updated any time nodes were added or removed from the cluster.
```

Comment:
can we not give the entire nodeCIDR range instead of an individual /32 list as the match?

Reply:
the problem is that, as described above, for ingress traffic at least, the traffic doesn't appear to come from the node CIDR, it comes from the node's `ovn-k8s-mp0` IP, which is inside the pod network CIDR. (There's no CIDR that selects "all node `mp0` IPs" without also selecting all pod IPs.)